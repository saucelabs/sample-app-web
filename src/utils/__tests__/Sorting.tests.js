import { sortAsc, sortDesc, sortHiLo, sortLoHi } from "../Sorting";

const data = [
  { id: 0, name: "Bar" },
  { id: 2.1, name: "Ar" },
  { id: 99.9, name: "Rab" },
];

it("should be able to sort an array with objects asc based on a property", () => {
  expect(sortAsc(data, "name")).toMatchSnapshot();
});

it("should be able to sort an array with objects desc based on a property", () => {
  expect(sortDesc(data, "name")).toMatchSnapshot();
});

it("should be able to sort an array with objects lo to high based on a property", () => {
  expect(sortLoHi(data, "id")).toMatchSnapshot();
});

it("should be able to sort an array with objects hi to low based on a property", () => {
  expect(sortHiLo(data, "id")).toMatchSnapshot();
});
