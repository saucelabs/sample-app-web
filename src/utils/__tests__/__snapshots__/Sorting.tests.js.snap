// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to sort an array with objects asc based on a property 1`] = `
Array [
  Object {
    "id": 2.1,
    "name": "Ar",
  },
  Object {
    "id": 0,
    "name": "Bar",
  },
  Object {
    "id": 99.9,
    "name": "Rab",
  },
]
`;

exports[`should be able to sort an array with objects desc based on a property 1`] = `
Array [
  Object {
    "id": 99.9,
    "name": "Rab",
  },
  Object {
    "id": 0,
    "name": "Bar",
  },
  Object {
    "id": 2.1,
    "name": "Ar",
  },
]
`;

exports[`should be able to sort an array with objects hi to low based on a property 1`] = `
Array [
  Object {
    "id": 99.9,
    "name": "Rab",
  },
  Object {
    "id": 2.1,
    "name": "Ar",
  },
  Object {
    "id": 0,
    "name": "Bar",
  },
]
`;

exports[`should be able to sort an array with objects lo to high based on a property 1`] = `
Array [
  Object {
    "id": 0,
    "name": "Bar",
  },
  Object {
    "id": 2.1,
    "name": "Ar",
  },
  Object {
    "id": 99.9,
    "name": "Rab",
  },
]
`;
